data/src/main/java/org/commcare/dalvik/data/network/NetworkUtil.kt

```diff
@@ -0,0 +1,72 @@
+package org.commcare.dalvik.data.network
+
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import com.google.gson.JsonObject
+import kotlinx.coroutines.flow.flow
+import org.commcare.dalvik.domain.model.AbdmErrorModel
+import org.commcare.dalvik.domain.model.HqResponseModel
+import retrofit2.Response
+import timber.log.Timber
+
+class NetworkUtil {
+    companion object {
+        const val BASE_URL = "https://ccind.duckdns.org/abdm/api/"
+        const val TRANSLATION_BASE_URL = "https://raw.githubusercontent.com/"
+        fun getTranslationEndpoint(code:String)=
+            "https://raw.githubusercontent.com/dimagi/abdm-app/main/resources/languages/${code}/language.json"
+    }
+}
+
+fun <T> safeApiCall(call: suspend () -> Response<T>) = flow {
+    this.emit(HqResponseModel.Loading)
+    try {
+        val response = call.invoke()
+        response.let {
+            var responseJsonObject:JsonObject
+            Timber.d("Network : ===> Response Received : code = ${response.code()}")
+            when(response.code()){
+                200 ->{
+                    it.body().toString().let {
+                        responseJsonObject = Gson().fromJson(it, JsonObject::class.java)
+                        if(responseJsonObject.has("code")){
+                            val adbmError:AbdmErrorModel =  Gson().fromJson(it, AbdmErrorModel::class.java)
+                            emit(HqResponseModel.AbdmError(500 , adbmError))
```

Comment:
Yes its success response , but while parsing the response we can have case when the response doesn't have the required field.
So ABDM internally emits an AbdmError which is handled in the UI layer and corresponding error message is displayed

This 500 is used kept for ABDM internal use only.

Reply:
> but while parsing the response we can have case when the response doesn't have the required field.

Curious what's the required field here ? Looking at the code it seems like we throw an error when there is `code` present in response instead of a field missing. What is this `code` used for and what can be different values for it ?